src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -110,4 +126,102 @@ private void addListingToBusinesses(PlacesSearchResult place) {
+    GeoApiContext context = new GeoApiContext.Builder()
+            .apiKey(KEY)
+            .build();
+    try {
+      for(int i = 0; i<allBusinesses.size();i++){
+        currentBusiness = allBusinesses.get(i);
+        TextSearchRequest request = new TextSearchRequest(context)
+                                            .query(currentBusiness.getName())
+                                            .location(latLng)
+                                            .radius(50000);
+        similarBusinessesInTheArea = request.await();
+        if(similarBusinessesInTheArea.results.length > 1){
+          checkBusinessThroughLinkedin(currentBusiness.getName());
+        }
+      }
+
+    }catch(Exception e) {
+      LOGGER.warning(e.getMessage());
+    }  
+  }
+  
+  private void checkBusinessThroughLinkedin(String currentBusinessName) 
+                          throws GeneralSecurityException, IOException {
+    String cx = "001390425498086947771:mhzyevhmmxq"; //Your search engine
+    //Instance Customsearch
+    Customsearch cs = new Customsearch.Builder(
+                                GoogleNetHttpTransport.newTrustedTransport(), 
+                                JacksonFactory.getDefaultInstance(), 
+                                null) 
+                                .setApplicationName("linkedinSearch") 
+                                .setGoogleClientRequestInitializer(
+                                      new CustomsearchRequestInitializer(KEY)) 
+                                .build();
+    Customsearch.Cse.List list = cs.cse().list(currentBusinessName).setCx(cx); 
+    List<Result> searchJsonResults = list.execute().getItems();
+    String[] numberOfFollowers;
+    if (searchJsonResults!=null){
+      if(searchJsonResults.size() != 0){
+        Result linkedinBusiness = searchJsonResults.get(0);
+        List<Map<String, Object>> resultsMetatags = 
+                        (List) linkedinBusiness.getPagemap().get("metatags");
+        for (Map<String, Object> tag : resultsMetatags) {
+          String b_title = (String) tag.get("og:description");
+          numberOfFollowers = b_title.split(" ");
+          findNumberOfFollowers(numberOfFollowers);
+        }
+      }
+    }
+  }
+
+  private void findNumberOfFollowers(String[] numberOfFollowers) {
+    int companyFollowers = 0;
+    for(int i = 0; i < numberOfFollowers.length; i++) {
+      if(numberOfFollowers[i].equals("followers")) {
+        companyFollowers = 
+                Integer.parseInt(numberOfFollowers[i-1].replaceAll(",", ""));
+      }
+    }
+    if(companyFollowers > minFollowers) {
+      addBigBusinessToDatabase();
+    }else {
+      checkNumberOfSimilarBusinessesInTheArea(currentBusiness.getName());
+    }
+  }
+
+  private void checkNumberOfSimilarBusinessesInTheArea(String businessName){
+    int countNumberOfMatchingBusiness = 0;
+    int i = 0;
+    while(i < similarBusinessesInTheArea.results.length 
```

Comment:
similarBusinessInTheArea is only relevant per business. Can you pass this as a parameter instead of making it global?

Reply:
I just made it a parameter, I made it global just because it is not used in every function